designs/per_command_help_text.md

```diff
@@ -0,0 +1,386 @@
+                                  permissions in your AWS account, for example, by creating new AWS Identity and
+                                  Access Management (IAM) users. For those stacks, you must explicitly acknowledge
+                                  their capabilities by specifying this parameter. The only valid values are
+                                  CAPABILITY_IAM and CAPABILITY_NAMED_IAM. If you have IAM resources, you can specify
+                                  either capability. If you have IAM resources with custom names, you must specify
+                                  CAPABILITY_NAMED_IAM. If you don't specify this parameter, this action returns an
+                                  InsufficientCapabilities error.
+                                  
+  Advanced Options:
+  
+  --code                          Sync code resources. This includes AWS Lambda Functions, API Gateway, and Step
+                                  Functions.
+
+  --watch                         Watch local files and automatically sync with remote.
+  --resource-id TEXT              Sync code for all the resources with the ID. To sync a resource within a nested
+                                  stack, use the following pattern {ChildStack}/{logicalId}.
+
+  --resource RESOURCE
+                                  Sync code for all resources of the given resource type. Accepted values are
+                                  ['AWS::Serverless::Function', 'AWS::Lambda::Function',
+                                  'AWS::Serverless::LayerVersion', 'AWS::Lambda::LayerVersion',
+                                  'AWS::Serverless::Api', 'AWS::ApiGateway::RestApi', 'AWS::Serverless::HttpApi',
+                                  'AWS::ApiGatewayV2::Api', 'AWS::Serverless::StateMachine',
+                                  'AWS::StepFunctions::StateMachine']
+
+  --dependency-layer / --no-dependency-layer
+                                  This option separates the dependencies of individual function into another layer,
+                                  for speeding up the sync.process
+
+
+
+  -s, --base-dir DIRECTORY        Resolve relative paths to function's source code with respect to this folder. Use
+                                  this if SAM template and your source code are not in same enclosing folder. By
+                                  default, relative paths are resolved with respect to the SAM template's location
+
+  -u, --use-container             If your functions depend on packages that have natively compiled dependencies, use
+                                  this flag to build your function inside an AWS Lambda-like Docker container
+
+                                  
+```
+
+
+Success criteria for the change
+-------------------------------
+
+* The success criteria for the change is less time spent on the help text to understand which options are useful.
```

Comment:
I think introducing some colour/sectioning will help with this a lot. Looking at the base help text, the bold and underlined sections make it a lot easier to quickly see what's going on. It directs a user's eyes to where they should be looking, instead of just seeing a massive wall of text.

Reply:
Definitely. All the required ones should have it. 

I'll try to add the NO_COLOR or --no-color along with this or scope it separately.